agent-shell.el

```diff
@@ -531,6 +544,36 @@ LINE defaults to 1, LIMIT defaults to nil (read to end)."
                            :code -32603
                            :message (error-message-string err))))))))
 
+(defun agent-shell--resolve-path (path)
+  "Resolve PATH using `agent-shell-path-resolver-function'."
+  (funcall (or agent-shell-path-resolver-function #'identity) path))
+
+(defun agent-shell--get-devcontainer-workspace-path (cwd)
+  "Return devcontainer workspaceFolder for CWD; signal error if none found."
+  (let ((devcontainer-config-file-name (expand-file-name ".devcontainer/devcontainer.json" cwd)))
+    (condition-case _err
+        (or
+         (map-elt (json-read-file devcontainer-config-file-name) 'workspaceFolder)
+         (error "No workspace folder defined in %s" devcontainer-config-file-name))
+      (file-missing (error "Not found: %s" devcontainer-config-file-name))
+      (permission-denied (error "Not readable: %s" devcontainer-config-file-name))
+      (json-string-format (error "No valid JSON: %s" devcontainer-config-file-name)))))
+
+(defun agent-shell--resolve-devcontainer-path (path)
+  "Resolve PATH from a devcontainer in the local filesystem, and vice versa."
```

Comment:
Would be nice to include an example usage so anyone looking to maintain this function gets a mental/visual picture of external behaviour.

"For example:

  (agent-shell--resolve-devcontainer-path "something")
    => "something else" "

Reply:
Added an example for both directions (container to local, and local to container) 👍